extra/intl-extra/IntlExtension.php

```diff
@@ -369,7 +369,7 @@ public function formatNumberStyle(string $style, $number, array $attrs = [], str
      */
     public function formatDateTime(Environment $env, $date, ?string $dateFormat = 'medium', ?string $timeFormat = 'medium', string $pattern = '', $timezone = null, string $calendar = 'gregorian', ?string $locale = null): string
     {
-        $date = CoreExtension::dateConverter($env, $date, $timezone);
+        $date = $env->getExtension(CoreExtension::class)->dateConverter($date, $timezone);
```

Comment:
this change requires bumping the min requirement of twig/twig in this intl-extra package.

Reply:
And btw, we probably need to keep a deprecated `dateConverter` static method to avoid issues if using twig/intl-extra 3.9.2 with twig/twig 3.10.0.
twig/intl-extra is using an internal API of a different package, which then becomes an issue for semver-based requirements.

Reply:
`twig/intl-extra` already raised it's twig requirement in the last release so it could just do that again?

Reply:
raising the requirement will solve my first comment. It will not solve the second one (which is about avoiding an error when using the old intl-extra with the new twig).

Reply:
3.9 broke the case of upgrading intl-extra alone by forgetting to bump the requirement. It did not broke the case of updating twig alone (as intl-extra 3.8 was using `twig_date_converter` which has not been removed)

Reply:
Done